corehq/apps/userreports/views.py

```diff
@@ -1366,6 +1357,19 @@ def rebuild_data_source(request, domain, config_id):
     ))
 
 
+def _prep_data_source_for_rebuild(data_source_config, is_static):
+    save_config = False
+    if not is_static:
+        data_source_config.meta.build.awaiting = True
+        save_config = True
+    if data_source_config.is_deactivated:
```

Comment:
When would a data source field `is_deactivated` be set? It's not clear to me why in the case the `is_deactivated` field is true, it would be able to rebuild? (I don't know the context behind the `is_deactivated` field)

Reply:
It seems a data source is deactivated if the corresponding report is removed or if the subscription for the project downgrades.

And we are setting it to be active again here in case its is rebuilt probably so that it can be used. It is existing implementation so I don't have context either.